plugins/vpp/natplugin/descriptor/nat44_vrf.go

```diff
@@ -0,0 +1,114 @@
+	nat "go.ligato.io/vpp-agent/v3/proto/ligato/vpp/nat"
+	"strconv"
+)
+
+const (
+	NAT44VrfDescriptorName = "vpp-nat44-vrf"
+
+	//Dependencies
+	nat44VrfTableDep = "vrf-table-exists"
+)
+
+// A list of non-retriable errors:
+var (
+// ErrNATInterfaceFeatureCollision is returned when VPP NAT features assigned
+// to a single interface collide.
+//ErrNATInterfaceFeatureCollision = errors.New("VPP NAT interface feature collision")
+
+// ErrDuplicateNATAddress is returned when VPP NAT address pool contains duplicate
+// IP addresses.
+// ErrDuplicateNATAddress = errors.New("Duplicate VPP NAT address")
+)
+
+// NAT44GlobalDescriptor teaches KVScheduler how to configure global options for
+// VPP NAT44.
+
+type NAT44VrfDescriptor struct {
+	log        logging.Logger
+	natHandler vppcalls.NatVppAPI
+
+	// UseDeprecatedAPI tracks whether deprecated global API (NAT interfaces, addresses) is being used on NB.
+	// Used to orchestrate which data should be dumped from which descriptor on Retrieve.
+	UseDeprecatedAPI bool
+}
+
+// NewNAT44GlobalDescriptor creates a new instance of the NAT44Global descriptor.
+func NewNAT44VrfDescriptor(natHandler vppcalls.NatVppAPI, log logging.PluginLogger) *kvs.KVDescriptor {
+	ctx := &NAT44VrfDescriptor{
+		natHandler: natHandler,
+		log:        log.NewLogger("nat44-vrf-descriptor"),
+	}
+	typedDescr := &adapter.NAT44VrfDescriptor{
+		Name:          NAT44VrfDescriptorName,
+		NBKeyPrefix:   nat.ModelNat44VrfTable.KeyPrefix(),
+		ValueTypeName: nat.ModelNat44VrfTable.ProtoName(),
+		KeySelector:   nat.ModelNat44VrfTable.IsKeyValid,
+		KeyLabel:      nat.ModelNat44VrfTable.StripKeyPrefix,
+		Create:        ctx.Create,
+		Delete:        ctx.Delete,
+		Retrieve:      ctx.Retrieve,
+	}
+	return adapter.NewNAT44VrfDescriptor(typedDescr)
+}
+
+// Create applies NAT44 global options.
+func (d *NAT44VrfDescriptor) Create(key string, vrfTable *nat.Nat44EdVrfTable) (metadata interface{}, err error) {
+	if !d.natHandler.WithLegacyStartupConf() {
+		err = d.natHandler.Nat44AddVrfTable(vrfTable.TableVrfId)
+		if err != nil {
+			d.log.Error(err)
+			return nil, err
+		}
+	}
+	return
+}
+
+// Delete sets NAT44 global options back to the defaults.
+func (d *NAT44VrfDescriptor) Delete(key string, vrfTable *nat.Nat44EdVrfTable, metadata interface{}) error {
+	err := d.natHandler.Nat44DelVrfTable(vrfTable.TableVrfId)
+	if err != nil {
+		d.log.Error(err)
+		return err
+	}
+	return nil
+}
+
+// Retrieve returns the current NAT44 interface configuration.
+func (d *NAT44VrfDescriptor) Retrieve(correlate []adapter.NAT44VrfKVWithMetadata) (
+	retrieved []adapter.NAT44VrfKVWithMetadata, err error) {
+
+	natTables, err := d.natHandler.Nat44VrfTablesDump()
+	if err != nil {
+		return nil, err
+	}
+	for _, natEntry := range natTables {
+		retrieved = append(retrieved, adapter.NAT44VrfKVWithMetadata{
+			//Key: nat.Nat44VrfTableKey(natEntry.TableVrfId),
+			Key:    strconv.Itoa(int(natEntry.TableVrfId)),
```

Comment:
Why not use the key creator for given modeled configuration (the above commented out line)? This is duplicity and can lead to errors as model definition (the name tepmlate) is the source of truth how the key should be formatted.